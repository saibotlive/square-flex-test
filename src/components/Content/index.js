import React, { useState } from 'react';
import {
  Container,
  Item,
  Image,
  Title,
  Body,
  LightBoxContainer,
  LightBoxBody,
  BodyStyle,
  LightBoxCloseBtn
} from './styles';
import useFetch from '../../hooks/useFetch';
import LightBox from '../LightBox';

const Content = () => {
  const data = useFetch('/api/getContent');
  const [lightbox, setLightBoxOpen] = useState({ item: null, isOpen: false });

  const handleLightBoxOpen = item => () => {
    setLightBoxOpen({ item, isOpen: true });
  };

  const handleLightBoxClose = () => {
    setLightBoxOpen({ item: null, isOpen: false });
  };

  return (
    <Container>
      {data.map(el => (
        <Item key={el.id} onClick={handleLightBoxOpen(el)}>
          <Image src={el.artwork} alt={el.title} />
          <Title>{el.title}</Title>
          <Body>{el.content}</Body>
        </Item>
      ))}
      {lightbox.isOpen && (
        <LightBox>
          <BodyStyle />
          <LightBoxContainer className="lightbox">
            <LightBoxBody>
              <div>
                <Image src={lightbox.item.artwork} alt={lightbox.item.title} />
              </div>
              <div />
              <Title>{lightbox.item.title}</Title>
              <Body>{lightbox.item.content}</Body>
            </LightBoxBody>
            <LightBoxCloseBtn className="close-btn" onClick={handleLightBoxClose} />
          </LightBoxContainer>
        </LightBox>
      )}
    </Container>
  );
};

export default Content;
