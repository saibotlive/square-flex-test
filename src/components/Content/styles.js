import styled, { createGlobalStyle, keyframes } from 'styled-components';

export const Container = styled.div`
  position: relative;
  width: 100%;
  /*display: flex;
  flex-wrap: wrap;
  flex: 0 1 auto;*/
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-gap: 10px 5px;
  padding: 0 5px;
`;

export const Item = styled.div`
  grid-column: span 12;

  &:nth-child(n + 10) {
    grid-column: span 6;
  }

  @media (min-width: 480px) {
    &:nth-child(1) {
      grid-column: span 12;
    }
    &:nth-child(n + 2):nth-child(-n + 9) {
      grid-column: span 6;
    }
    &:nth-child(n + 10):nth-child(-n + 15) {
      grid-column: span 4;
    }
  }

  @media (min-width: 1024px) {
    &:nth-child(1) {
      grid-column: span 8;
    }

    &:nth-child(n + 2):nth-child(-n + 5) {
      grid-column: span 4;
    }

    &:nth-child(n + 6):nth-child(-n + 9) {
      grid-column: span 3;
    }

    &:nth-child(n + 10):nth-child(-n + 15) {
      grid-column: span 2;
    }
  }
`;

export const Image = styled.img`
  display: block;
  max-width: 100%;
  height: auto;
`;

export const Title = styled.div`
  font-size: 16px;
  font-weight: bold;
  margin: 22px;
`;

export const Body = styled.p`
  font-size: 14px;
  margin: 22px;
`;

export const BodyStyle = createGlobalStyle`
  html,
  body {
    overflow: hidden;
    width: 100%;
    height: 100%;
  }
`;

const fadeIn = keyframes`
  from {
    opacity: 0;
  }

  to {
    opacity: 1;
  }
`;

export const LightBoxContainer = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.7);
  animation: ${fadeIn} 0.5s linear;
`;

export const LightBoxBody = styled.div`
  position: absolute;
  background-color: #ffffff;
  display: block;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 50%;
  max-width: 800px;

  & > div {
    font-size: 14px;
    font-weight: bold;
    margin: 16px;
  }
  p {
    font-size: 10px;
    margin: 16px;
  }

  @media (min-width: 960px) {
    width: 75%;
    & > div {
      font-size: 16px;
      font-weight: bold;
      margin: 22px;
    }
    ${Body} {
      font-size: 14px;
      margin: 22px;
    }
  }
`;

export const LightBoxItem = styled.div`
  position: relative;
`;

export const LightBoxCloseBtn = styled.div`
  position: absolute;
  cursor: pointer;
  right: 16px;
  top: 16px;
  width: 32px;
  height: 32px;
  opacity: 0.7;
  transition: opacity 0.3s;

  &:hover {
    opacity: 1;
  }

  &:before,
  &:after {
    position: absolute;
    left: 15px;
    content: ' ';
    height: 33px;
    width: 2px;
    background-color: #fff;
  }

  &:before {
    transform: rotate(45deg);
  }
  &:after {
    transform: rotate(-45deg);
  }
`;
