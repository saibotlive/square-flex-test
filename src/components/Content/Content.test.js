import React from 'react';
import { render, fireEvent, waitForDomChange, cleanup, waitForElement } from 'react-testing-library';
import { act } from 'react-dom/test-utils';
import Content from '../Content';

describe('<Content/>', () => {
  afterEach(cleanup);

  it('should render content', async () => {
    const { container } = render(<Content />);
    act(() => {
      waitForDomChange(container);
    });

    expect(container).toMatchSnapshot();
  });

  it('should show lightbox', async () => {
    const { container } = render(<Content />);

    const item = await waitForElement(() => container.querySelectorAll('div')[1]);
    fireEvent.click(item);

    const lightbox = document.body.querySelector('.lightbox');
    expect(lightbox).toMatchSnapshot();
  });

  it('should remove lightbox', async () => {
    const { container } = render(<Content />);

    const item = await waitForElement(() => container.querySelectorAll('div')[1]);
    fireEvent.click(item);

    const lightbox = document.body.querySelector('.lightbox');

    const closeBtn = lightbox.querySelector('.close-btn');
    fireEvent.click(closeBtn);
    waitForDomChange(document.body);
    expect(document.body).toMatchSnapshot();
  });
});
