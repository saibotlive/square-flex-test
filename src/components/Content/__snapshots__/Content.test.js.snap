// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content/> should render content 1`] = `
<div>
  <div
    class="sc-bdVaJa lEnjB"
  />
</div>
`;

exports[`<Content/> should show lightbox 1`] = `
<div
  class="lightbox sc-EHOje jKFcDn"
>
  <div
    class="sc-bZQynM kfTwdK"
  >
    <div>
      <img
        alt="Lorem Ipsum 1"
        class="sc-htpNat juBNep"
        src="https://placehold.it/1920x1080"
      />
    </div>
    <div />
    <div
      class="sc-bxivhb gSrOLL"
    >
      Lorem Ipsum 1
    </div>
    <p
      class="sc-ifAKCX rlLby"
    >
      1. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat
    </p>
  </div>
  <div
    class="close-btn sc-htoDjs kmknyU"
  />
</div>
`;
